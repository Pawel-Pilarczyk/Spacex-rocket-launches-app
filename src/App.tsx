import React from 'react';
import {ApolloProvider, ApolloClient, InMemoryCache} from '@apollo/client';
import RootNavigation from './navigation';

export const API = 'https://api.spacex.land/graphql';

const client = new ApolloClient({
  uri: API,
  cache: new InMemoryCache(),
});

const App = () => {
  return (
    <ApolloProvider client={client}>
      <RootNavigation />
    </ApolloProvider>
  );
};

export default App;
