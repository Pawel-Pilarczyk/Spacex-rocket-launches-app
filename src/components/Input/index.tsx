import {TextInput, View, ViewStyle, StyleSheet} from 'react-native';
import React from 'react';
import {scaling} from 'src/styles/scaling';
import {COLORS} from 'src/constants/colors';
import {MagnifyingGlass} from 'src/assets/svg';

type TProps = {
  type?: 'text' | 'search';
  value?: string;
  setValue: (str: string) => void;
  style?: ViewStyle | ViewStyle[];
  placeholder?: string;
  testID?: string;
};

const Input = ({
  setValue,
  type = 'text',
  value,
  style,
  placeholder,
  testID,
}: TProps) => {
  return (
    <View style={[style, styles.wrapper]}>
      {type === 'search' && <MagnifyingGlass testID="magnifyingGlass" />}
      <TextInput
        onChangeText={setValue}
        value={value}
        style={styles.textInput}
        autoFocus={false}
        maxLength={100}
        placeholder={placeholder}
        testID={testID}
      />
    </View>
  );
};

export default Input;

const styles = StyleSheet.create({
  wrapper: {
    height: scaling.hs(48),
    backgroundColor: COLORS.WHITE,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingHorizontal: scaling.hs(10),
    borderWidth: 1,
    borderColor: COLORS.PRIMARY,
  },
  textInput: {
    flex: 1,
  },
});
