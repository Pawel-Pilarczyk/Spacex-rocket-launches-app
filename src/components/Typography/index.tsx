import {StyleProp, Text, TextStyle} from 'react-native';
import React from 'react';
import {scaling} from 'src/styles/scaling';
import {COLORS} from 'src/constants/colors';

type TSize =
  | '8'
  | '10'
  | '12'
  | '14'
  | '16'
  | '18'
  | '22'
  | '24'
  | '28'
  | '32'
  | '40';
export type TFontType = 'bold' | 'regular' | 'light';

const fontWeight: Record<TFontType, any> = {
  bold: '800',
  regular: '400',
  light: '200',
};

type TProps = {
  children: string | Element | Element[];
  color?: string;
  size?: TSize;
  type?: TFontType;
  style?: StyleProp<TextStyle> | StyleProp<TextStyle>[];
};

const Typography = ({
  size = '12',
  type = 'regular',
  children,
  style,
  color = COLORS.BLACK,
}: TProps) => {
  return (
    <Text
      style={[
        style,
        {
          fontSize: scaling.hs(+size),
          fontWeight: fontWeight[type],
          color,
        },
      ]}>
      {children}
    </Text>
  );
};

export default Typography;
