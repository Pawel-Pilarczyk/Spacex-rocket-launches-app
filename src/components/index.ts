export {default as Button} from './Button';
export {default as Typography} from './Typography';
export {default as Input} from './Input';
export {default as Item} from './Item';
export {default as ModalLoader} from './Modals/ModalLoader';
