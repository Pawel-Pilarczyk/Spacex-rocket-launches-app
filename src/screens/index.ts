export {default as Home} from './Home';
export {default as Welcome} from './Welcome';
export {default as LaunchDetails} from './LaunchDetails';
export {default as ErrorScreen} from './Error';
