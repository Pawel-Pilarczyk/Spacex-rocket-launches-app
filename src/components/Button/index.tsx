import {StyleProp, StyleSheet, TouchableOpacity, ViewStyle} from 'react-native';
import React from 'react';
import {Typography} from 'src/components';
import {scaling} from 'src/styles/scaling';
import {COLORS} from 'src/constants/colors';

type TButtonType = 'primary' | 'outline';

type TProps = {
  onPress?: () => void;
  type?: TButtonType;
  children: string;
  style?: StyleProp<ViewStyle>;
  disabled?: boolean;
  testID?: string;
};

const Button = ({
  children,
  onPress,
  type = 'primary',
  style,
  disabled,
  testID,
}: TProps) => {
  const handlePress = () => {
    if (!disabled && onPress) {
      onPress();
    }
  };
  return (
    <TouchableOpacity
      testID={testID}
      onPress={handlePress}
      disabled={disabled}
      style={[
        style,
        styles.wrapper,
        type === 'primary' && styles.primary,
        type === 'outline' && styles.outline,
        disabled && styles.disabled,
      ]}>
      <Typography
        size="18"
        color={type === 'primary' || disabled ? COLORS.WHITE : COLORS.PRIMARY}>
        {children}
      </Typography>
    </TouchableOpacity>
  );
};

export default Button;

const styles = StyleSheet.create({
  wrapper: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    height: scaling.vs(56),
    maxHeight: scaling.vs(56),
    width: '100%',
    borderRadius: scaling.vs(16),
  },
  primary: {
    backgroundColor: COLORS.PRIMARY,
  },
  outline: {
    backgroundColor: COLORS.WHITE,
    borderColor: COLORS.PRIMARY,
    borderWidth: 2,
  },
  disabled: {
    backgroundColor: COLORS.DISABLED,
    borderColor: COLORS.DISABLED,
  },
});
