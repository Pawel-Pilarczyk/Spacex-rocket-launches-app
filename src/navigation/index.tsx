import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {ROUTES} from 'src/constants/routes';
import {Home, Welcome, LaunchDetails, ErrorScreen} from 'src/screens';

const RootStack = createStackNavigator();

const RootNavigation = () => {
  return (
    <NavigationContainer>
      <RootStack.Navigator
        screenOptions={{
          headerTitleAlign: 'center',
          headerBackTitle: '',
        }}>
        <RootStack.Screen
          name={ROUTES.WELCOME}
          component={Welcome}
          options={{headerShown: false}}
        />
        <RootStack.Screen
          name={ROUTES.HOME}
          component={Home}
          options={{headerTitle: 'Missions List'}}
        />
        <RootStack.Screen
          name={ROUTES.DETAILS}
          component={LaunchDetails}
          options={{headerTitle: 'Mission Details'}}
        />
        <RootStack.Screen name={ROUTES.ERROR} component={ErrorScreen} />
      </RootStack.Navigator>
    </NavigationContainer>
  );
};

export default RootNavigation;
