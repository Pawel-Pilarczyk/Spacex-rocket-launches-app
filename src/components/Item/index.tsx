import React from 'react';
import {View, StyleSheet, TouchableOpacity} from 'react-native';
import {TLaunch} from 'src/types/index';
import {COLORS} from 'src/constants/colors';
import {scaling} from 'src/styles/scaling';
import {Typography} from 'src/components';
import {RocketFailure, RocketSuccess, Arrow} from 'src/assets/svg';

type Props = {
  data: TLaunch;
  onPress: () => void;
};

const Item = ({data, onPress}: Props) => {
  const {mission_name, launch_success} = data;
  return (
    <TouchableOpacity onPress={onPress} testID="itemID">
      <View
        style={[
          styles.wrapper,
          launch_success ? styles.lineSuccess : styles.lineError,
        ]}>
        <View style={styles.titleWrapper}>
          {launch_success ? (
            <RocketSuccess width={25} height={25} testID="rocketSuccess" />
          ) : (
            <RocketFailure width={25} height={25} testID="rocketFailure" />
          )}
          <Typography
            color={COLORS.WHITE}
            size="14"
            type="bold"
            style={styles.spacer}>
            {mission_name}
          </Typography>
        </View>
        <Arrow testID="arrowIcon" />
      </View>
    </TouchableOpacity>
  );
};

export default Item;

const styles = StyleSheet.create({
  wrapper: {
    height: scaling.vs(50),
    borderColor: COLORS.PRIMARYDARK,
    borderWidth: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingHorizontal: scaling.hs(10),
    marginVertical: scaling.vs(5),
  },
  titleWrapper: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  lineSuccess: {
    backgroundColor: COLORS.SUCCESS,
  },
  lineError: {
    backgroundColor: COLORS.ERROR,
  },
  spacer: {
    marginLeft: scaling.hs(10),
  },
});
