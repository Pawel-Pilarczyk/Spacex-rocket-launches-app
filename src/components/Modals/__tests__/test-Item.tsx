import 'react-native';
import React from 'react';
import {render} from '@testing-library/react-native';
import {ModalLoader} from 'src/components';

it('Should render correctly', () => {
  const modalLoader = render(<ModalLoader />);
  expect(modalLoader).toMatchSnapshot();
});

it('Should dipslay "Loading" text', () => {
  const modalLoader = render(<ModalLoader />);
  const text = modalLoader.findByText('Loading');
  expect(text).toBeDefined();
});

it('Should have a loader child', () => {
  const modalLoader = render(<ModalLoader />);
  const loader = modalLoader.findByTestId('loader');
  expect(loader).toBeDefined();
});
