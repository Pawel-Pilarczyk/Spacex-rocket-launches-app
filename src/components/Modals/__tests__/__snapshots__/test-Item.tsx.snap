// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  testID="modalLoader"
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(256, 256, 256, 0.9)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      color="#1C6DD0"
      size="large"
      testID="loader"
    />
    <Text
      style={
        Array [
          undefined,
          Object {
            "color": "#000",
            "fontSize": 48,
            "fontWeight": "400",
          },
        ]
      }
    >
      Loading
    </Text>
  </View>
</Modal>
`;
